src/psyclone/tests/psyad/domain/lfric/test_lfric_adjoint.py

```diff
@@ -57,81 +68,543 @@ def test_generate_lfric_adjoint_no_container_error(fortran_reader):
   var1 = var2
 end program test
 """)
-    with pytest.raises(InternalError) as err:
+    with pytest.raises(GenerationError) as err:
         generate_lfric_adjoint(psyir, ["var1", "var2"])
-    assert ("An LFRic kernel must be within a Container but the supplied "
-            "PSyIR does not contain one." in str(err.value))
+    assert ("If the LFRic kernel contains a single container, it should not "
+            "be a FileContainer (as that means the kernel does not contain "
+            "a module)." in str(err.value))
 
 
 def test_generate_lfric_adjoint_no_routines_error(fortran_reader):
     '''
     Check that generate_lfric_adjoint raises the expected error when
-    provided PSyIR that does not contain any Routines.
+    provided with PSyIR that does not contain any Routines.
 
     '''
     psyir = fortran_reader.psyir_from_source("""\
-module test
+module test_mod
   implicit none
   integer :: var1
-end module test
+end module test_mod
 """)
     with pytest.raises(InternalError) as err:
         generate_lfric_adjoint(psyir, ["var1", "var2"])
     assert "The supplied PSyIR does not contain any routines" in str(err.value)
 
 
-def test_generate_lfric_adjoint_multi_kernel(fortran_reader, fortran_writer):
```

Comment:
Have you removed this as we no longer support multiple kernel routines in a file?

Reply:
Yes, when using the new metadata processing approach it does not currently support multi-precision kernels so I dropped the multi-kernel test. I am working towards multi-precision support, it is just that the metadata values currently disappear in PSyIR so I need to work on that first before tackling the fact that metadata multi-precision kernels are treated as structures not unknown fortran type (as they don't have a contains/procedure), so need parsing separately.

Reply:
OK.